src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/MicrosoftNetCoreAnalyzersResources.resx

```diff
@@ -2132,4 +2132,19 @@ Widening and user defined conversions are not supported with generic types.</val
   <data name="DoNotPassNullableStructToArgumentNullExceptionThrowIfNullCodeFixTitle" xml:space="preserve">
     <value>Replace the 'ArgumentNullException.ThrowIfNull' call with a conditional</value>
   </data>
+  <data name="DoNotCompareSpanToNullDescription" xml:space="preserve">
+    <value>Because the 'null' literal will be implicitly converted to a span, comparing a span to 'null' might not do what you intended to check. Make it more explicit by comparing to 'default' or by using 'IsEmpty'.</value>
+  </data>
+  <data name="DoNotCompareSpanToNullMessage" xml:space="preserve">
+    <value>Do not compare Span&lt;T&gt; to 'null', use 'default' or check 'IsEmpty' instead</value>
```

Comment:
```suggestion
    <value>Do not compare Span&lt;T&gt; to 'null'; compare to 'default' or check 'IsEmpty' instead</value>
```

Reply:
With the new requirement now we need to remove the suggestion for default, I updated @gewarren's suggestion accordingly: 
```suggestion
    <value>Do not compare Span&lt;T&gt; to 'null'; remove the comparison as struct will never be 'null' or check 'IsEmpty' that was the intention</value>
```